services/cd-service/pkg/repository/transformer.go

```diff
@@ -637,7 +637,7 @@ func (c *CreateApplicationVersion) Transform(
 			return "", err
 		}
 		t.AddAppEnv(c.Application, env, teamOwner)
-		if ((hasUpstream && config.Upstream.Latest) || slices.Contains(c.DeployToDownstreamEnvironments, env)) && isLatest && !c.IsPrepublish {
+		if ((hasUpstream && config.Upstream.Latest && isLatest) || slices.Contains(c.DeployToDownstreamEnvironments, env)) && !c.IsPrepublish {
```

Comment:
I think we can extract some variables here to make this more readable, something like this:
```suggestion
        envIsConfiguredLatest:=hasUpstream && config.Upstream.Latest && isLatest
        downstreamDeploymentRequested := slices.Contains(c.DeployToDownstreamEnvironments, env)
		if (envIsConfiguredLatest || downstreamDeploymentRequested) && !c.IsPrepublish {
```